packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -677,11 +677,6 @@ public void stopBackgroundThread() {
     if (backgroundHandlerThread != null) {
       stoppingBackgroundHandlerThread = true;
       backgroundHandlerThread.quitSafely();
-      try {
-        backgroundHandlerThread.join();
```

Comment:
What does `stoppingBackgroundHandlerThread` do without this join? Presumably from the description `quitSafely` is asynchronous, so it looks like we're now setting it to true and then back to false within the same execution cycle.

Reply:
Oh you're right. I think I added that not knowing it was acting as a bandaid for the real problem!